lib/events/sessionlog.go

```diff
@@ -49,10 +52,58 @@ type SessionLogger interface {
 	WriteChunk(chunk *SessionChunk) (written int, err error)
 }
 
-// diskSessionLogger implements a disk based session logger. The imporant
+// DiskSessionLoggerConfig sets up parameters for disk session logger
+// associated with the session ID
+type DiskSessionLoggerConfig struct {
+	// SessionID is the session id of the logger
+	SessionID session.ID
+	// EventsFileName is the events file name
+	EventsFileName string
+	// StreamFileName is the byte stream file name
+	StreamFileName string
+	// Clock is the clock replacement
+	Clock clockwork.Clock
+}
+
+// NewDiskSessionLogger creates new disk based session logger
+func NewDiskSessionLogger(cfg DiskSessionLoggerConfig) (*DiskSessionLogger, error) {
+	writtenBytes, err := ReadWrittenBytes(cfg.EventsFileName)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// create a new session stream file:
+	fstream, err := os.OpenFile(cfg.StreamFileName, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0640)
```

Comment:
Is there a constant for `0640`? Below too.

Reply:
I will handle this in the next PR in the module